src/west/manifest.py

```diff
@@ -87,6 +87,14 @@
 
 _logger = logging.getLogger(__name__)
 
+# Type for the submodule value passed through the manifest file.
+class Submodule(NamedTuple):
+    name: str
+    path: str
```

Comment:
So I started actually trying this new feature and got confused by the purpose of the `name` atrribute. Then I realized why I got confused: `name` is not anywhere.
Then I asked whether `name` could be removed... while  west 0.9 was released. Too late?

So in #474 I'm making `name` optional, please help review and/or test.

PS: `name` is barely used in `git submodule` too.